rpc/namespaces/ethereum/eth/api.go

```diff
@@ -2,198 +2,312 @@ package eth
+	e.logger.Debug("eth_getStorageAt", "address", address.Hex(), "key", key, "block number or hash", blockNrOrHash)
+	return e.backend.GetStorageAt(address, key, blockNrOrHash)
+}
+
+// GetCode returns the contract code at the given address and block number.
+func (e *PublicAPI) GetCode(address common.Address, blockNrOrHash rpctypes.BlockNumberOrHash) (hexutil.Bytes, error) {
+	e.logger.Debug("eth_getCode", "address", address.Hex(), "block number or hash", blockNrOrHash)
+	return e.backend.GetCode(address, blockNrOrHash)
+}
+
+// GetProof returns an account object with proof and any storage proofs
+func (e *PublicAPI) GetProof(address common.Address, storageKeys []string, blockNrOrHash rpctypes.BlockNumberOrHash) (*rpctypes.AccountResult, error) {
+	e.logger.Debug("eth_getProof", "address", address.Hex(), "keys", storageKeys, "block number or hash", blockNrOrHash)
+	return e.backend.GetProof(address, storageKeys, blockNrOrHash)
+}
+
+///////////////////////////////////////////////////////////////////////////////
+///                           EVM/Smart Contract Execution				          ///
+///////////////////////////////////////////////////////////////////////////////
+
+// Call performs a raw contract call.
+func (e *PublicAPI) Call(args evmtypes.TransactionArgs, blockNrOrHash rpctypes.BlockNumberOrHash, _ *rpctypes.StateOverride) (hexutil.Bytes, error) {
+	e.logger.Debug("eth_call", "args", args.String(), "block number or hash", blockNrOrHash)
+
+	blockNum, err := e.backend.GetBlockNumber(blockNrOrHash)
 	if err != nil {
-		return "", err
+		return nil, err
+	}
+	data, err := e.backend.DoCall(args, blockNum)
+	if err != nil {
+		return []byte{}, err
 	}
-	ethAddr := common.BytesToAddress(coinbase.Bytes())
-	return ethAddr.Hex(), nil
-}
 
-// Mining returns whether or not this node is currently mining. Always false.
-func (e *PublicAPI) Mining() bool {
-	e.logger.Debug("eth_mining")
-	return false
+	return (hexutil.Bytes)(data.Ret), nil
 }
 
-// Hashrate returns the current node's hashrate. Always 0.
-func (e *PublicAPI) Hashrate() hexutil.Uint64 {
-	e.logger.Debug("eth_hashrate")
-	return 0
+///////////////////////////////////////////////////////////////////////////////
+///                           Event Logs													          ///
+///////////////////////////////////////////////////////////////////////////////
+// FILTER API
+
+///////////////////////////////////////////////////////////////////////////////
+///                           Chain Information													          ///
+///////////////////////////////////////////////////////////////////////////////
+
+// ProtocolVersion returns the supported Ethereum protocol version.
+func (e *PublicAPI) ProtocolVersion() hexutil.Uint {
+	e.logger.Debug("eth_protocolVersion")
+	return hexutil.Uint(ethermint.ProtocolVersion)
 }
 
 // GasPrice returns the current gas price based on Ethermint's gas price oracle.
```

Comment:
@facs95 Any reason, why you didn't move this endpoint logic?

Reply:
any method of `PublicAPI` gets exposed as an endpoint (`eth_protocolVersion`) its not the same when its a backend method. Those are not exposed. So I just moved the logic that use clientQuery or clientCtx to the backend. Anything that required state query.